File: cs/Minsk.Tests/CodeAnalysis/Text/SourceTextTests.cs
using Minsk.CodeAnalysis.Text;
using Xunit;

namespace Minsk.Tests.CodeAnalysis.Text;

public sealed class SourceTextTests
{
    [Theory]
    [InlineData(".", 1)]
    [InlineData(".\r\n", 2)]
    [InlineData(".\r\n\r\n", 3)]
    private void ReportsCorrectLineCount(string text, int expectedLineCount)
    {
        var sourceText = SourceText.From(text);
        Assert.Equal(expectedLineCount, sourceText.Lines.Length);
    }
}

File: kt/minsk/src/test/kotlin/dev/phytolizer/minsk/analysis/text/SourceTextTests.kt
package dev.phytolizer.minsk.analysis.text

import io.kotest.core.spec.style.FunSpec
import io.kotest.data.forAll
import io.kotest.data.row
import io.kotest.matchers.shouldBe

@Suppress("unused")
class SourceTextTests : FunSpec({
    test("SourceText includes last line") {
        forAll(
            row(".", 1),
            row(".\r\n", 2),
            row(".\r\n\r\n", 3),
        ) { text, expectedLineCount ->
            val sourceText = SourceText.from(text)
            sourceText.lines.size shouldBe expectedLineCount
        }
    }
})
